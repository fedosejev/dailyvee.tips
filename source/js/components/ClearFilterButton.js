import React, { Component } from 'react';

class ClearFilterButton extends Component {
  handleClick = () => {
    const { onClearFilter } = this.props;

    onClearFilter();
  }

  render() {
    return (
      <button className="clearFilterButton" title="Clear filter" onClick={this.handleClick}>
        <i className="fa fa-times-circle" aria-hidden="true" />
      </button>
    );
  }
}

ClearFilterButton.propTypes = {
  onClearFilter: React.PropTypes.func.isRequired,
};

export default ClearFilterButton;
