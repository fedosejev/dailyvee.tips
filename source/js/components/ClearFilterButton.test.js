import React from 'react';
import ClearFilterButton from './ClearFilterButton';
import renderer from 'react-test-renderer';

describe('ClearFilterButton', () => {
  test('SHOULD render correctly', () => {
    const tree = renderer.create(
      <ClearFilterButton />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
