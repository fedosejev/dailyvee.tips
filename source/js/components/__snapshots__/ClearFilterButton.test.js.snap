// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearFilterButton SHOULD render correctly 1`] = `
<button
  className="clearFilterButton"
  onClick={[Function]}
  title="Clear filter"
>
  <i
    aria-hidden="true"
    className="fa fa-times-circle"
  />
</button>
`;
